o2a/mappers/action_mapper.py

```diff
@@ -39,20 +44,21 @@ def __init__(self, oozie_node: Element, name: str, dag_name: str, props: Propert
 
     def on_parse_node(self, workflow: Workflow):
         super().on_parse_node(workflow)
-        self._parse_config()
+        self._parse_config(workflow)
+
+    def _parse_config(self, workflow: Workflow):
+        job_xml_nodes = find_nodes_by_tag(self.oozie_node, TAG_JOB_XML)
+        new_properties = extract_properties_from_job_xml_nodes(
+            job_xml_nodes=job_xml_nodes, input_directory_path=workflow.input_directory_path, props=self.props
```

Comment:
If I understand correctly we are only using the `Workflow` in mappers to extract the `input_directory_path `?
If so, then I'd suggest passing only this variable and not the whole workflow object. This way we wouldn't couple `ActionMapper` and `Workflow` so tightly.

Reply:
I updated the PR. I passed input_directory_path via constructor.